rfc-029-source-code-should-be-open-source-by-default.md

```diff
@@ -0,0 +1,145 @@
+
+Allowing the team to work in the open also has benefits for individual
+developers. They:
+
+- can show their work to friends and family—show the thing instead of
+  talking about the thing
+- know that they’re investing their energy in something maximally
+  useful and long-lasting
+- can build up a public portfolio of work
+- can talk about their work without worrying about restrictions like
+  NDAs
+
+We should also make a simultaneous decision about software
+licensing. Whilst theoretically possible to make a GitHub repository
+public without an explicit license, we shouldn’t do this, because:
+
+- having a public repository without a licence means that legally
+  [nobody can do anything useful with it](https://choosealicense.com/no-permission/),
+  so there’s not much point in doing so
+- releasing software without explicit no-liability, no-warranty
+  clauses might cause legal problems for us one day
+
+We had a brief discussion about this as a team and the MIT Licence is
+the one that came up the most. It’s a simple licence which allows
+people to do what they wish with the code and removes any potential
+legal issues for us. The other licence that came up is some version of
+the [GPL](https://choosealicense.com/licenses/gpl-3.0/), but even if
+it were something we wished to use, we wouldn’t have the resources to
+monitor for and pursue license violations.
+
+## Proposal
+
+All new GitHub respositories in the dxw organisation SHOULD be public,
+unless they directly relate to a client project. When these
+repositories are public, they SHOULD contain a copy of the [MIT
+License](https://choosealicense.com/licenses/mit/) in a `LICENSE` file
+in the repository’s root.
+
+We MAY allow a non-client repository to be private if it contains:
+
+- secrets such as keys—although this is bad practice and should not
+  have happened in the first place
+- personal information
+- commercially sensitive information
+
+If we choose to make such a repository private, it MUST contain a
+`WHY_PRIVATE.md` document explaining why it’s private, with reference
+to one of these reasons. This is useful because:
+
+- it allows us to one day revisit the decision to make the repository
+  private, without relying on team members’ memories
+- we can set up automated checks to ensure that all private
+  repositories contain this justification
+
+When starting a new client project, we SHOULD ask the client whether
+we can make the repository public. We SHOULD explain the benefits of
+making it public.  Where relevant, we can refer to existing guidance
+such as the [Government Design
+Principles](https://www.gov.uk/guidance/government-design-principles#make-things-open-it-makes-things-better)
+or the [GOV.UK Service
+Standard](https://www.gov.uk/service-manual/service-standard/point-12-make-new-source-code-open).
+
+This proposal doesn’t address things that aren’t hosted on
+GitHub—mainly our GovPress work.
```

Comment:
Why not? We need to justify, under this, why the govpress sites shouldn't move to github and be open source. It may be just because of time and effort, which is fine, but in that case I'd say remove this line and we will just have to make things conform over time when we can.

Reply:
Ah, I see that's in the "next steps" section. I'd say remove this paragraph if that's the long-term intention, or at least add the word `currently`.

Reply:
I think it's because I didn't want to turn it into a conversation about where the GovPress repositories should be hosted. I've removed the sentence and lengthened the "next step".